### PR TITLE
Use shallow merge for frontendPreferences

### PR DESCRIPTION
**Pull request by davidmz**
Closes T69

### CODE REVIEW

File: test/functional/users.js [hunk 1]
```diff
@@ -943,28 +946,24 @@ describe("UsersController", function() {
...
-          data.users.frontendPreferences['net.freefeed'].should.have.deep.property('screenName.displayOption')
-          data.users.frontendPreferences['net.freefeed'].screenName.displayOption.should.equal(2)
-          data.users.frontendPreferences['net.freefeed'].should.have.deep.property('screenName.useYou')
-          data.users.frontendPreferences['net.freefeed'].screenName.useYou.should.equal(false)
+          data.users.frontendPreferences['net.freefeed'].should.be.deep.equal(newPrefs['net.freefeed'])
           // custom domain
           data.users.frontendPreferences.should.have.property('custom.domain')
-          data.users.frontendPreferences['custom.domain'].should.have.property('customProperty')
-          data.users.frontendPreferences['custom.domain'].customProperty.should.equal('someWeirdValue')
+          data.users.frontendPreferences['custom.domain'].should.be.deep.equal(newPrefs['custom.domain'])
```
Comment (clbn) [inline test/functional/users.js]: This check is nice, but it seems it just can be replaced with `.deep.equal(newPrefs)` then, which is not what we want. So to highlight this case, I'd really like to see another domain in `prefs` which is NOT PRESENT in `newPrefs`, so we can make sure it doesn't disappear.

Comment (clbn) [inline test/functional/users.js]: (Before your change "custom.domain" did that, but now it plays different role, so we need a new actor for that.)

Comment (clbn) [inline test/functional/users.js]: Actually, maybe we should leave "custom.domain" its old role, and just introduce a new domain to test shallow merging.

**Comment by davidmz** [comment]
OK, I added a more comprehensive test.

**Pull request merged**